internal/pkg/groups/fillcache.go

```diff
@@ -93,6 +93,27 @@ func (c *FillCache) Update(group string) (bool, error) {
 	return false, err
 }
 
+// update wraps the Update method, adding instrumentation
+func (c *FillCache) update(group string) {
```

Comment:
Very open to any thoughts around making this new method 'fit in' a little bit more

Reply:
we say this "wraps" the `Update` method -- but I don't see any wrapping here?

Reply:
🤔 Would it be better to just add this instrumentation to the `Update` method?

Reply:
I did think that, but the original setup felt pretty nice to me - to have the instrumentation separated. Happy to move it into the `Update` method if you think that'd be cleaner still.

Reply:
Looking again, maybe it's not so bad? I'll give it a quick try and see how it fits.

Reply:
Added some potential changes to bring some of the instrumentation into the `Update` method -- it now only returns a `bool`, instead of a `bool` and `error`. The error seemed to only be used _for_ the instrumentation, though it was being tested against in some capacity (purely when testing the functionality, not in 'production'. 

Maybe there's a reason we want to keep returning those errors though